bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/MyRenaultHttpSession.java

```diff
@@ -0,0 +1,266 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.renault.internal.renault.api;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.renault.internal.RenaultConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Translation of this python code: https://github.com/hacf-fr/renault-api
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+public class MyRenaultHttpSession {
+
+    private HttpClient httpClient;
+    private Constants constants;
+
+    private Car car;
+    private String kamereonToken;
+    private String kamereonaccountId;
+    private String cookieValue;
+    private String personId;
+    private String gigyaDataCenter;
+    private String jwt;
+
+    private final Logger logger = LoggerFactory.getLogger(MyRenaultHttpSession.class);
+
+    public MyRenaultHttpSession(RenaultConfiguration config) throws Exception {
+
+        car = new Car();
+        this.httpClient = new HttpClient(new SslContextFactory(true));
+        this.constants = new Constants(config.locale);
+
+        httpClient.start();
+        login(config);
+        getAccountInfo(config);
+        getJWT(config);
+        getAccountID(config);
+        getVehicle(config);
+    }
+
+    public void updateCarData(RenaultConfiguration config) throws Exception {
+        getCockpit(config);
+        getBatteryStatus(config);
+        getLocation(config);
+        getHvacStatus(config);
+    }
+
+    private void login(RenaultConfiguration config) throws Exception {
+
+        Fields fields = new Fields();
+        fields.add("ApiKey", this.constants.getGigyaApiKey());
+        fields.add("loginID", config.myRenaultUsername);
+        fields.add("password", config.myRenaultPassword);
+
+        logger.debug("URL: {}/accounts.login", this.constants.getGigyaRootUrl());
+        ContentResponse response = this.httpClient.FORM(this.constants.getGigyaRootUrl() + "/accounts.login", fields);
+        if (HttpStatus.OK_200 == response.getStatus()) {
+            JsonObject responseJson = new JsonParser().parse(response.getContentAsString()).getAsJsonObject();
+            JsonObject sessionInfoJson = responseJson.getAsJsonObject("sessionInfo");
+            cookieValue = sessionInfoJson.get("cookieValue").getAsString();
+            logger.debug("Cookie: {}", cookieValue);
+        } else {
+            logger.error("Response: [{}] {}\n{}", response.getStatus(), response.getReason(),
+                    response.getContentAsString());
```

Comment:
Done set to warning.  Not all features are supported so some stuff will not work.  I would like to be verbose about this so the user has a chance to understand that the reason it does not work is because the server said "no".